JeffreyC iOS6 Submission

JeffreyBytes opened the pull request.

JeffreyBytes committed:
Create Initial Xcode Project

JeffreyBytes committed:
Added custom Question type and QuestionController

JeffreyBytes committed:
Added TableView, Custom Cell, and Ask Question UI

Added custom Question Cell
Added custom Question View Controller

JeffreyBytes committed:
Added AnswerViewController UI

JeffreyBytes committed:
Implemented tableview data source methods

JeffreyBytes committed:
Properly Implement Segues Between ViewControllers

JeffreyBytes committed:
Fixed autolayout issue in AnswerVC

JeffreyBytes committed:
Implemented ability to delete questions

JeffreyBytes committed:
Update TextViews to be 1/3 of height of the super view

JeffreyBytes committed:
Update AutoLayout Constraints to remove build warnings

nelglez commented:
Great job Jeffrey, the app has proper constraints and everything works well. In your answer view controller I just made the following modifications. What you did is fine, we just generally do something like this.

```
 var question: Question? {
        didSet {
            //Every time there is a change call updateViews()
            updateViews()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        //we need to call updateViews() here as well.
        updateViews()

//        if let question = question {
//            questionLabel.text = question.question
//            askerLabel.text = question.asker
//        }
        
    }
    
    private func updateViews() {
        //Unwrap question so we are certain it contains a value otherwise return out of the function. Make sure the view has been loaded too or it can creash the app.
        guard let question = question, isViewLoaded else {return}
        questionLabel.text = question.question
        askerLabel.text = question.asker
    }
```

Also another way to check that something is not empty is ```guard let text = !text.isEmpty``` 

Awesome work!

JeffreyBytes committed:
Removed boilerplate

JeffreyBytes committed:
Refactored Answer and AskQuestion VCs

AnswerVC and AskQuestionVC now utilize the isEmpty method build into String.
AnswerVC refactored to include PM's suggestions.
